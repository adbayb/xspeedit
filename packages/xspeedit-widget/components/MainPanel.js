import React, { Component, Fragment } from "react";
import Packager, { optimizedStrategy, actualStrategy } from "xspeedit-core";

import Chart from "../components/Chart";
import Input from "../components/Input";

class MainPanel extends Component {
	state = {
		optimizedBoxes: [],
		unoptimizedBoxes: []
	};
	packager = new Packager("", 10);

	handleInputSubmit = input => {
		this.packager.input = input;
		this.setState(
			() => ({
				optimizedBoxes: this.packager.setStrategy(optimizedStrategy).resolve(),
				unoptimizedBoxes: this.packager.setStrategy(actualStrategy).resolve()
			}),
			() => {
				console.log(this.state, "STATE");
			}
		);
		console.log("isInputValid: here result", input, this.packager);
	};

	isInputValid = input => {
		return input && input.length > 0 && /^\d+$/.test(input);
	};

	render() {
		const { optimizedBoxes, unoptimizedBoxes } = this.state;

		return (
			<Fragment>
				<Input label="Objets" onValidate={this.isInputValid} onSubmit={this.handleInputSubmit} />
				{unoptimizedBoxes &&
					unoptimizedBoxes.length > 0 && (
						<Chart
							key="main-actual"
							title={`Rangement non optimisé: ${unoptimizedBoxes.length} carton(s)`}
							data={unoptimizedBoxes}
						/>
					)}
				{optimizedBoxes &&
					optimizedBoxes.length > 0 && (
						<Chart
							key="main-optimized"
							title={`Rangement optimisé: ${optimizedBoxes.length} carton(s)`}
							data={optimizedBoxes}
						/>
					)}
			</Fragment>
		);
	}
}

export default MainPanel;
