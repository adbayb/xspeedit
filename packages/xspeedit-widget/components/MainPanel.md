**Preview:**

```js
<div style={{ backgroundColor: "black", height: 400 }}>
	<MainPanel />
</div>;
```

**Example:**

```jsx static
import React from "react";

const Example = () => {
	return (
		<MainPanel />
	);
};
```
