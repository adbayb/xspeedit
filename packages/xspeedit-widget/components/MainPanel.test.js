import React from "react";
import MainPanel from "./MainPanel.js";

describe("MainPanel", () => {
	it("should render correctly", () => {
		// GIVEN: notre composant:
		const tree = shallow(<MainPanel />);
		const instance = tree.instance();
		// @note: issue Fragment avec enzyme (https://github.com/airbnb/enzyme/issues/1213):
		const fragment = instance.render();
		// WHEN: nous envoyons un input:
		instance.handleInputSubmit("0123456789");
		// THEN: il doit s'afficher correctement en matchant son snapshot:
		expect(shallow(<div>{fragment}</div>).getElement()).toMatchSnapshot();
	});

	it("should check input correctly", () => {
		// GIVEN: notre composant:
		// WHEN: nous appellons le validateur d'input:
		// THEN: il doit s'afficher correctement en matchant son snapshot:
		expect(MainPanel.isInputValid("012364444")).toBe(true);
	});
});
