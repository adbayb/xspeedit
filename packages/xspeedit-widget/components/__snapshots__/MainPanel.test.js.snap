// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPanel should render correctly 1`] = `
<MainPanel>
  <Input
    label="Objets"
    onSubmit={[Function]}
    onValidate={[Function]}
  >
    <form
      className="jsx-2110388131"
      onSubmit={[Function]}
    >
      <label
        className="jsx-2110388131"
        htmlFor="boxesInput"
      >
        Objets
      </label>
      <input
        className="jsx-2110388131"
        id="boxesInput"
        name="input"
        onChange={[Function]}
        placeholder="Saisissez vos objets (taille comprise entre 1 et 9)"
        type="number"
        value=""
      />
      <button
        className="jsx-2110388131"
        disabled={true}
        type="submit"
      >
        <Icon
          className="button"
          height={24}
          name="enter"
          width={24}
        >
          <img
            alt="icon-enter"
            className="button"
            height={24}
            src="/static/enter.svg"
            width={24}
          />
        </Icon>
      </button>
      <JSXStyle
        css="form.jsx-2110388131{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:none;-ms-flex:none;flex:none;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;background-color:#6c727f;}input.jsx-2110388131{width:calc(100% - 50px);font-size:0.9rem;margin:0 0.5rem;-webkit-appearance:none;-moz-appearance:textfield;-webkit-appearance:none;-moz-appearance:none;appearance:none;border:0;background-color:transparent;color:white;}input.jsx-2110388131::-webkit-input-placeholder{color:currentColor;}input.jsx-2110388131::-moz-placeholder{color:currentColor;}input.jsx-2110388131:-ms-input-placeholder{color:currentColor;}input.jsx-2110388131::placeholder{color:currentColor;}button.jsx-2110388131{padding:0 20px;background:none;border:none;border-radius:0;}button.jsx-2110388131:not([disabled]){background-color:#34b7b6;}label.jsx-2110388131{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;background-color:#565b68;margin:0;padding:1rem;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}"
        styleId="2110388131"
      />
    </form>
  </Input>
  <section
    className="jsx-4012226680"
  >
    <Icon
      height={100}
      name="chevrons-up"
      width={100}
    >
      <img
        alt="icon-chevrons-up"
        height={100}
        src="/static/chevrons-up.svg"
        width={100}
      />
    </Icon>
    <JSXStyle
      css="section.jsx-4012226680{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex:1;-ms-flex:1;flex:1;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}section.jsx-4012226680 img{-webkit-animation-duration:0.75s;animation-duration:0.75s;-webkit-animation-name:slideup-jsx-4012226680;animation-name:slideup-jsx-4012226680;-webkit-animation-direction:alternate;animation-direction:alternate;-webkit-animation-iteration-count:infinite;animation-iteration-count:infinite;}@-webkit-keyframes slideup-jsx-4012226680{from{-webkit-transform:translate3d(0,100px,0);-ms-transform:translate3d(0,100px,0);transform:translate3d(0,100px,0);}to{-webkit-transform:translate3d(0,50px,0);-ms-transform:translate3d(0,50px,0);transform:translate3d(0,50px,0);}}@keyframes slideup-jsx-4012226680{from{-webkit-transform:translate3d(0,100px,0);-ms-transform:translate3d(0,100px,0);transform:translate3d(0,100px,0);}to{-webkit-transform:translate3d(0,50px,0);-ms-transform:translate3d(0,50px,0);transform:translate3d(0,50px,0);}}"
      styleId="4012226680"
    />
  </section>
</MainPanel>
`;
